sources/sns/subscription.go

```diff
@@ -0,0 +1,113 @@
+package sns
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/sns"
+	"github.com/overmindtech/aws-source/sources"
+	"github.com/overmindtech/sdp-go"
+)
+
+type subsCli interface {
+	GetSubscriptionAttributes(ctx context.Context, params *sns.GetSubscriptionAttributesInput, optFns ...func(*sns.Options)) (*sns.GetSubscriptionAttributesOutput, error)
+	ListSubscriptions(context.Context, *sns.ListSubscriptionsInput, ...func(*sns.Options)) (*sns.ListSubscriptionsOutput, error)
+	ListTagsForResource(context.Context, *sns.ListTagsForResourceInput, ...func(*sns.Options)) (*sns.ListTagsForResourceOutput, error)
+}
+
+func getSubsFunc(ctx context.Context, client subsCli, scope string, input *sns.GetSubscriptionAttributesInput) (*sdp.Item, error) {
+	output, err := client.GetSubscriptionAttributes(ctx, input)
+	if err != nil {
+		return nil, err
+	}
+
+	if output.Attributes == nil {
+		return nil, &sdp.QueryError{
+			ErrorType:   sdp.QueryError_NOTFOUND,
+			ErrorString: "get subscription attributes response was nil",
+		}
+	}
+
+	attributes, err := sources.ToAttributesCase(output.Attributes)
+	if err != nil {
+		return nil, err
+	}
+
+	resourceTags, err := tagsByResourceARN(ctx, client, *input.SubscriptionArn)
+	if err != nil {
+		return nil, &sdp.QueryError{
+			ErrorType:   sdp.QueryError_NOTFOUND,
+			ErrorString: err.Error(),
+		}
+	}
+
+	item := &sdp.Item{
+		Type:            "sns-subscription",
+		UniqueAttribute: "subscriptionArn",
+		Attributes:      attributes,
+		Scope:           scope,
+		Tags:            tagsToMap(resourceTags),
+	}
+
+	topicArn, err := attributes.Get("topicArn")
+	if err != nil {
+		return nil, err
+	}
+
+	if topicArn.(string) != "" {
+		// +overmind:link sns-topic
+		item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+			Query: &sdp.Query{
+				Type:   "sns-topic",
+				Method: sdp.QueryMethod_GET,
+				Query:  topicArn.(string),
+				Scope:  scope,
+			},
+			BlastPropagation: &sdp.BlastPropagation{
+				// If topic is not healthy, subscription will not work
+				In: true,
+				// Subscription won't affect the topic
+				Out: false,
+			},
+		})
+	}
```

Comment:
We should also be able to link `SubscriptionRoleArn` to an IAM role if it exists

Reply:
https://github.com/overmindtech/aws-source/pull/415/commits/e47b3d93b03a1e3f4120730ba5979e5de54d54f3